simulator/src/data/backupCircuit.js

```
@@ -42,30 +45,43 @@ export function backUp(scope = globalScope) {
+    for (let i = 0; i < scope.SubCircuit.length; i++) {
+        scope.SubCircuit[i].makeConnections();
+    }
 
     return data;
 }
 
 export function scheduleBackup(scope = globalScope) {
     var backup = JSON.stringify(backUp(scope));
-    if (scope.backups.length === 0 || scope.backups[scope.backups.length - 1] !== backup) {
+    if (
+        scope.backups.length === 0 ||
```

ESLint

[error] '||' should be placed at the beginning of the line.

(operator-linebreak)